src/Compiler/Facilities/AsyncMemoize.fs

```diff
@@ -534,6 +534,17 @@ type internal AsyncMemoize<'TKey, 'TVersion, 'TValue when 'TKey: equality and 'T
 
         }
 
+    member _.TryGet(key) =
+        let x = cache.TryGet(key)
+
+        match x with
+        | Some(Completed(value, _)) -> Some value
+        | _ -> None
+
+    member _.Set(key, value) =
```

Comment:
This seems quite wrong. The way to set something here would probably be to call `.Get` with a completed computation, so it goes through all the synchronization mechanisms. 

But also directly setting things is not really how this is supposed to be used.